bgflow/nn/flow/crd_transform/ic.py

```diff
@@ -235,103 +138,128 @@ class ReferenceSystemTransformation(Flow):
     ----------
     normalize_angles : bool
         bring angles and torsions into (0, 1) interval
+    orientation : "euler" | "basis"
+        which representation is used to represent the global orientation of the system
     eps : float
         numerical epsilon used to enforce manifold boundaries
     raise_warnings : bool
         raise warnings if manifold boundaries are violated
     """
 
-    def __init__(self, normalize_angles=True, eps=1e-7, enforce_boundaries=True, raise_warnings=True):
+    def __init__(
+        self,
+        normalize_angles=True,
+        eps=1e-7,
+        enforce_boundaries=True,
+        raise_warnings=True,
+    ):
         super().__init__()
         self._normalize_angles = normalize_angles
         self._eps = eps
         self._enforce_boundaries = enforce_boundaries
         self._raise_warnings = raise_warnings
 
     def _forward(self, x0, x1, x2, *args, **kwargs):
+        """
+        Parameters:
+        ----------
+        x0, x1, x2: torch.Tensor
+            xyz coordinates of the first three points
 
-        R = orientation(
-            x0,
-            x1,
-            x2,
-            eps=self._eps,
-            enforce_boundaries=self._enforce_boundaries,
-            raise_warnings=self._raise_warnings
-        )
-        d01, _ = dist_deriv(
-            x0,
-            x1,
-            eps=self._eps,
-            enforce_boundaries=self._enforce_boundaries,
-            raise_warnings=self._raise_warnings
-        )
-        d12, _ = dist_deriv(
-            x1,
-            x2,
-            eps=self._eps,
-            enforce_boundaries=self._enforce_boundaries,
-            raise_warnings=self._raise_warnings
-        )
-        a012, _ = angle_deriv(
+        Returns:
+        --------
+        x0: torch.Tensor
+            origin of the system
+        orientation: torch.Tensor
+            if orientation is "basis" returns [3,3] matrix 
+            if orientation is "euler" returns tuple with three Euler angles (alpha, beta, gamma)
+                their range are in [0, pi] if unnormalized and [0, 1] if normalized
```

Comment:
[0,pi]?
If I understand you correctly, beta is always in [0,1]?
And it looks like you are calling `normalize_torsions` rather than `normalize_angles` on alpha and gamma. Are you sure they are not in [0, 2pi] or [-pi, pi]?